### DISCUSSION TITLE
Is there any way to affect all objects in a gameobject list?

### DISCUSSION THREAD

**Discussion by AustiNNowers** [community/community#68095] [Programming Help] [answered]
### Body

I'm making a game, so I created a list of gameobjects, in the same code I made two functions to make objects that have a collider possible to drag the object around the scene, but I don't know how to incorporate these functions into all the objects that are added to list.

```
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class MovimentoJogador : MonoBehaviour
{
    public List<GameObject> jogador = new();
    
    Vector2 diferenca = Vector2.zero;

    private void OnMouseDown()
    {
        diferenca = (Vector2)Camera.main.ScreenToWorldPoint(Input.mousePosition) - (Vector2)transform.position;
    }

    private void OnMouseDrag()
    {
        transform.position = (Vector2)Camera.main.ScreenToWorldPoint(Input.mousePosition) - diferenca;
    }
}
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by rantirules** [comment, marked as answer]
I believe if you want to apply the drag functionality to all the game objects in the jogador list you are going to have to iterate through the list.

```cpp
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class MovimentoJogador : MonoBehaviour
{
    public List<GameObject> jogador = new();
    
    Vector2 diferenca = Vector2.zero;

    private void Start()
    {
        foreach (GameObject obj in jogador)
        {
            obj.AddComponent<DraggableObject>();
        }
    }
}

```
And then you'll have to modify your original code so that it contains a new script like DraggableObject which will include the OnMouseDown and OnMouseDrag functions and then just attach this to any to each of the objects you want to have the draggable functionality.
```cpp
using UnityEngine;

public class DraggableObject : MonoBehaviour
{
    Vector2 diferenca = Vector2.zero;

    private void OnMouseDown()
    {
        diferenca = (Vector2)Camera.main.ScreenToWorldPoint(Input.mousePosition) - (Vector2)transform.position;
    }

    private void OnMouseDrag()
    {
        transform.position = (Vector2)Camera.main.ScreenToWorldPoint(Input.mousePosition) - diferenca;
    }
}

```

**Reply by AustiNNowers** [reply]
Thank you soooo much, this is perfect.
